stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveData.kt

```diff
@@ -74,10 +76,22 @@ internal class MessageListItemLiveData(
         typingLd?.let { typing -> configTypingChange(typing, currentUser) }
     }
 
+    /**
+     * Emits a value from this [MediatorLiveData] class when either
+     * the user gets updated or the deleted message visibility gets
+     * changed.
+     */
     private fun configMessagesChange(messages: LiveData<List<Message>>, getCurrentUser: LiveData<User?>) {
-        val messagesChange = getCurrentUser.changeOnUserLoaded(messages) { changedMessages, currentUser ->
-            messagesChanged(changedMessages, currentUser!!.id)
-        }
+        val messagesChange = getCurrentUser
+            .combineWith(deletedMessageVisibility) { user, _ ->
+                user
+            }
+            .changeOnUserLoaded(messages) { changedMessages, currentUser ->
+                messagesChanged(
+                    changedMessages,
+                    currentUser!!.id
```

Comment:
We could filter the user in case it's null or its ID is to avoid potentially crashing the app

Reply:
Added a null check in an if statement checking that the user is not null.